pkg/apiserver-impl/api_default_service.go

```diff
@@ -2,62 +2,78 @@ package apiserver_impl
 
 import (
 	"context"
-	"errors"
-	openapi "github.com/redhat-developer/odo/pkg/apiserver-gen/go"
+	"fmt"
 	"net/http"
+
+	openapi "github.com/redhat-developer/odo/pkg/apiserver-gen/go"
+	"github.com/redhat-developer/odo/pkg/component/describe"
+	"github.com/redhat-developer/odo/pkg/kclient"
+	odocontext "github.com/redhat-developer/odo/pkg/odo/context"
+	"github.com/redhat-developer/odo/pkg/podman"
+	"github.com/redhat-developer/odo/pkg/state"
 )
 
 // DefaultApiService is a service that implements the logic for the DefaultApiServicer
 // This service should implement the business logic for every endpoint for the DefaultApi API.
 // Include any external packages or services that will be required by this service.
 type DefaultApiService struct {
+	cancel       context.CancelFunc
+	pushWatcher  chan<- struct{}
+	kubeClient   kclient.ClientInterface
+	podmanClient podman.Client
+	stateClient  state.Client
 }
 
 // NewDefaultApiService creates a default api service
-func NewDefaultApiService() openapi.DefaultApiServicer {
-	return &DefaultApiService{}
+func NewDefaultApiService(
+	cancel context.CancelFunc,
+	pushWatcher chan<- struct{},
+	kubeClient kclient.ClientInterface,
+	podmanClient podman.Client,
+	stateClient state.Client,
+) openapi.DefaultApiServicer {
+	return &DefaultApiService{
+		cancel:       cancel,
+		pushWatcher:  pushWatcher,
+		kubeClient:   kubeClient,
+		podmanClient: podmanClient,
+		stateClient:  stateClient,
+	}
 }
 
 // ComponentCommandPost -
 func (s *DefaultApiService) ComponentCommandPost(ctx context.Context, componentCommandPostRequest openapi.ComponentCommandPostRequest) (openapi.ImplResponse, error) {
-	// TODO - update ComponentCommandPost with the required logic for this service method.
-	// Add api_default_service.go to the .openapi-generator-ignore to avoid overwriting this service implementation when updating open api generation.
-
-	// TODO: Uncomment the next line to return response Response(200, GeneralSuccess{}) or use other options such as http.Ok ...
-	// return Response(200, GeneralSuccess{}), nil
-
-	return openapi.Response(http.StatusNotImplemented, nil), errors.New("ComponentCommandPost method not implemented")
+	switch componentCommandPostRequest.Name {
+	case "push":
+		s.pushWatcher <- struct{}{}
+		return openapi.Response(http.StatusOK, nil), nil
+	default:
+		return openapi.Response(http.StatusBadRequest, nil), fmt.Errorf("command name %q not supported. Supported values are: %q", componentCommandPostRequest.Name, "push")
```

Comment:
Wondering if the response body in case of error should not be serialized as a real JSON string too. Currently, I get a simple string, which does not respect the response content type returned in the response headers:

```
$ http POST http://127.0.0.1:20002/api/v1/component/command name=fake-cmd
HTTP/1.1 400 Bad Request
Content-Length: 74
Content-Type: application/json; charset=UTF-8
Date: Thu, 22 Jun 2023 15:45:12 GMT

"command name \"fake-cmd\" not supported. Supported values are: \"push\""
```

I'd expect something like below. What do you think?

```json
{
  "message": "command name \"fake-cmd\" not supported. Supported values are: \"push\""
}
```

Reply:
I think we will need to complete the openapi spec, to point to the correct `GeneralError` instead of using the default. WDYT?

Reply:
Yes, that makes sense.